cg/apps/lims/api.py

```diff
@@ -322,20 +323,32 @@ def _most_recent_date(self, dates: list):
 
     def _get_methods(self, step_names_udfs, lims_id):
         """
-        Gets the method, method number and method version for a given list of stop names
+        Gets the method, method number and method version for a given list of step names for AM documents.
+        Only method name and Atlas version is returned if Atlas documentation instead has been used.
         """
         methods = []
 
         for process_name in step_names_udfs:
             artifacts = self.get_artifacts(process_type=process_name, samplelimsid=lims_id)
-            if artifacts:
-                udf_key_number = step_names_udfs[process_name]["method_number"]
-                udf_key_version = step_names_udfs[process_name]["method_version"]
+            if not artifacts:
+                continue
+            # Get a list of parent processes for the artifacts
+            processes = self.get_processes_from_artifacts(artifacts=artifacts)
+            for process in processes:
+                # Check which type of method document has been used
+                method_type = self.get_method_type(process, step_names_udfs[process_name])
+                if method_type == DocumentationMethod.ATLAS:
+                    udf_key_method_doc = step_names_udfs[process_name]["atlas_document"]
+                    udf_key_version = step_names_udfs[process_name]["atlas_version"]
+                else:
+                    udf_key_method_doc = step_names_udfs[process_name]["method_number"]
+                    udf_key_version = step_names_udfs[process_name]["method_version"]
```

Comment:
I think this look up should be a function as is it traverses 2 levels of of a container i.e. it requires knowledge on what the internals of step_names_udfs looks like

Reply:
I added a new function for these lines, get_method_udf_values. Was it this you meant?